# Performance Metrics

Objective: load match-level player performance data, filter to Brazilian
Série A 2025 matches, and aggregate into season totals per player.

In [8]:
import pandas as pd

## Loading games and appearances data

In [9]:
df_games = pd.read_csv('../data/raw/games.csv')

In [10]:
# carregue appearances.csv, salve em df_appearances
df_appearances = pd.read_csv('../data/raw/appearances.csv')

## Exploring structure

In [11]:
df_games.columns

Index(['game_id', 'competition_id', 'season', 'round', 'date', 'home_club_id',
       'away_club_id', 'home_club_goals', 'away_club_goals',
       'home_club_position', 'away_club_position', 'home_club_manager_name',
       'away_club_manager_name', 'stadium', 'attendance', 'referee', 'url',
       'home_club_formation', 'away_club_formation', 'home_club_name',
       'away_club_name', 'aggregate', 'competition_type'],
      dtype='str')

In [12]:
df_appearances.columns

Index(['appearance_id', 'game_id', 'player_id', 'player_club_id',
       'player_current_club_id', 'date', 'player_name', 'competition_id',
       'yellow_cards', 'red_cards', 'goals', 'assists', 'minutes_played'],
      dtype='str')

In [13]:
df_games['season'].unique()

array([2009, 2005, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019,
       2020, 2021, 2025, 2022, 2024, 2023, 2007])

## Filtering to Brazilian Série A 2025

In [18]:
df_games_brazil = df_games[(df_games['competition_id'] == 'BRA1') & (df_games['season'] == 2025)]
df_games_brazil

,game_id,competition_id,season,round,date,home_club_id,away_club_id,home_club_goals,away_club_goals,home_club_position,...,stadium,attendance,referee,url,home_club_formation,away_club_formation,home_club_name,away_club_name,aggregate,competition_type
88297,4810877,BRA1,2025,18. Matchday,2026-05-30,210,199,1,3,16.0,...,Arena do Grêmio,42359.0,Lucas Paulo Torezin,https://www.transfermarkt.co.uk/gremio-foot-ba...,4-2-3-1,4-2-3-1,Grêmio Foot-Ball Porto Alegrense,Sport Club Corinthians Paulista,1:3,domestic_league
88298,4810878,BRA1,2025,18. Matchday,2026-05-30,679,3876,1,0,4.0,...,Estádio Mário Celso Petraglia,27927.0,Fernando Antonio Mendes,https://www.transfermarkt.co.uk/club-athletico...,3-4-2-1,4-2-3-1,Clube Atlético Paranaense,Mirasol Futebol Clube,1:0,domestic_league
88299,4810879,BRA1,2025,18. Matchday,2026-05-30,10010,537,2,1,6.0,...,Arena Fonte Nova,25249.0,Davi de Oliveira Lacerda,https://www.transfermarkt.co.uk/esporte-clube-...,4-3-3 Attacking,4-2-3-1,Esporte Clube Bahia,S. A. F. Botafogo,2:1,domestic_league
88300,4810880,BRA1,2025,18. Matchday,2026-06-01,10997,585,1,0,18.0,...,Estádio Estadual Jornalista Edgar Augusto Proença,25649.0,Jonathan Benkenstein Pinheiro,https://www.transfermarkt.co.uk/clube-do-remo-...,4-2-3-1,4-4-2,Clube do Remo,São Paulo Futebol Clube,1:0,domestic_league
88301,4810881,BRA1,2025,18. Matchday,2026-05-31,8793,6600,3,1,5.0,...,Estádio Municipal Cicero De Souza Marques,5559.0,Alex Gomes Stefano,https://www.transfermarkt.co.uk/red-bull-braga...,4-2-3-1,3-4-2-1,Red Bull Bragantino,Sport Club Internacional,3:1,domestic_league
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
88469,4811052,BRA1,2025,15. Matchday,2026-05-10,330,537,1,1,13.0,...,Arena MRV,33043.0,Ramon Abatti Abel,https://www.transfermarkt.co.uk/clube-atletico...,4-2-3-1,3-4-3,Clube Atlético Mineiro,S. A. F. Botafogo,1:1,domestic_league
88470,4811053,BRA1,2025,15. Matchday,2026-05-10,210,614,0,1,17.0,...,Arena do Grêmio,30891.0,Davi de Oliveira Lacerda,https://www.transfermarkt.co.uk/gremio-foot-ba...,3-4-3,4-2-3-1,Grêmio Foot-Ball Porto Alegrense,Clube de Regatas do Flamengo,0:1,domestic_league
88471,4811054,BRA1,2025,15. Matchday,2026-05-09,776,6600,2,2,9.0,...,Estádio Major Antônio Couto Pereira,15878.0,Wagner do Nascimento Magalhães,https://www.transfermarkt.co.uk/coritiba-foot-...,4-2-3-1,5-4-1,Coritiba Foot Ball Club,Sport Club Internacional,2:2,domestic_league
88472,4811055,BRA1,2025,15. Matchday,2026-05-10,10010,609,1,2,6.0,...,Arena Fonte Nova,30423.0,Rodrigo José Pereira de Lima,https://www.transfermarkt.co.uk/esporte-clube-...,4-1-4-1,4-2-3-1,Esporte Clube Bahia,Cruzeiro Esporte Clube,1:2,domestic_league


In [19]:
df_games_brazil['round'].unique()

<StringArray>
['18. Matchday', '17. Matchday', '16. Matchday', '14. Matchday',
 '13. Matchday', '12. Matchday', '11. Matchday', '10. Matchday',
  '9. Matchday',  '8. Matchday',  '7. Matchday',  '6. Matchday',
  '5. Matchday',  '4. Matchday',  '3. Matchday',  '2. Matchday',
  '1. Matchday', '15. Matchday']
Length: 18, dtype: str

In [21]:
df_games_brazil['date'].min()

'2026-01-28'

In [22]:
df_games_brazil['date'].max()

'2026-06-01'

In [ ]:
df_appearances_brazil = df_appearances[df_appearances['game_id'].isin(df_games_brazil['game_id'])]
df_appearances_brazil

,appearance_id,game_id,player_id,player_club_id,player_current_club_id,date,player_name,competition_id,yellow_cards,red_cards,goals,assists,minutes_played


In [28]:
df_appearances_brazil.shape

(0, 13)

In [30]:
df_games_brazil['game_id'].dtype

dtype('int64')

In [31]:
df_appearances['game_id'].dtype

dtype('int64')

In [32]:
df_games_brazil.shape

(177, 23)

In [33]:
df_games_brazil['game_id'].head()

88297    4810877
88298    4810878
88299    4810879
88300    4810880
88301    4810881
Name: game_id, dtype: int64

In [34]:
df_appearances['game_id'].head()

0    2231978
1    2233748
2    2234413
3    2234418
4    2234421
Name: game_id, dtype: int64

In [35]:
df_appearances['game_id'].min()

np.int64(2211607)

In [36]:
df_appearances['game_id'].max()

np.int64(4868900)

In [37]:
df_games_brazil['game_id'].min()

np.int64(4810877)

In [38]:
df_games_brazil['game_id'].max()

np.int64(4811056)

In [39]:
df_games_brazil['game_id'].isin(df_appearances['game_id']).sum()

np.int64(0)

In [40]:
df_appearances[df_appearances['competition_id'] == 'BRA1'].shape

(0, 13)

In [41]:
df_appearances[df_appearances['competition_id'] == 'BRA1']['date'].max()

nan

In [42]:
df_appearances['competition_id'].unique()

<StringArray>
[ 'CLQ',  'ELQ', 'UKRS',  'RUP', 'UKR1',  'DK1', 'RUSS',  'RU1', 'BESC',
 'UKRP',  'BE1', 'FRCH', 'POCP',  'SC1', 'NLSC',  'CIT',  'FR1',  'NL1',
  'SCI', 'POSU',  'DFL', 'GBCS',  'DFB',  'TR1',  'PO1',  'GB1',  'ES1',
  'SUC',   'L1',  'GR1',  'IT1',  'CDR',  'DKP',  'USC',   'CL',   'EL',
  'NLP',  'SFA',  'GRP',  'FAC', 'KLUB', 'ECLQ',  'CGB', 'AFCN', 'POBE',
 'EJPL', 'BPO4', 'FIWC']
Length: 48, dtype: str

## Conclusion

- games.csv has full coverage for Brazilian Série A (season 2025: 177 matches
  up to matchday 18, out of 38; season 2024: full season available)
- appearances.csv (match-level individual player stats) does NOT cover
  Brazilian competitions — investigation of unique competition_id values
  confirmed it only includes European leagues (GB1, ES1, IT1, FR1, L1, etc.)
- Decision: individual performance stats will be sourced from a second
  dataset (eduardopalmieri/brasileiro-player-stats-2024), merged by player
  name in the next notebook
- Project will realign to season 2024 across all sources (players, games,
  performance stats) for temporal consistency